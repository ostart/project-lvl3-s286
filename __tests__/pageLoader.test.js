import fs from 'mz/fs';
import os from 'os';
import nock from 'nock';
import path from 'path';

import pageLoad from '../src';


describe('Page Loader', () => {
  beforeAll(() => {
    nock.disableNetConnect();
  });

  it('test data', async () => {
    const host = 'http://ru.hexlet.io';
    const body = 'test data';
    nock(host).get('/').reply(200, body);

    // создать временную директорию
    const tempDir = await fs.mkdtemp(`${os.tmpdir()}${path.sep}`);
    const response = await pageLoad(host, tempDir);
    expect(response).toBe(true);
    // проверить что есть файл во временной директории с именем ru-hexlet-io.html
    const fileName = 'ru-hexlet-io.html';
    const isFileExists = await fs.exists(path.join(tempDir, fileName));
    expect(isFileExists).toBe(true);
    // проверить что содержимое файла 'test data'
    const fileContent = await fs.readFile(path.join(tempDir, fileName), 'utf-8');
    expect(fileContent).toBe(body);
  });
});
