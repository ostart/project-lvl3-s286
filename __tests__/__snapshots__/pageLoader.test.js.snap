// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Loader test 404 for resource files 1`] = `[Error: Request failed with status code 404]`;
